Gemfile

```diff
@@ -100,21 +100,35 @@ end
 # BULLET TRAIN GEMS
 # This section is the list of Ruby gems included by default for Bullet Train.
 
+# We use a constant here so that we can ensure that all of the bullet_train-*
+# packages are on the same version.
+BULLET_TRAIN_VERSION = "1.3.24"
+
 # Core packages.
-gem "bullet_train"
-gem "bullet_train-super_scaffolding"
-gem "bullet_train-api"
-gem "bullet_train-outgoing_webhooks"
-gem "bullet_train-incoming_webhooks"
-gem "bullet_train-themes"
-gem "bullet_train-themes-light"
-gem "bullet_train-integrations"
-gem "bullet_train-integrations-stripe"
+gem "bullet_train", BULLET_TRAIN_VERSION
+gem "bullet_train-super_scaffolding", BULLET_TRAIN_VERSION
+gem "bullet_train-api", BULLET_TRAIN_VERSION
+gem "bullet_train-outgoing_webhooks", BULLET_TRAIN_VERSION
+gem "bullet_train-incoming_webhooks", BULLET_TRAIN_VERSION
+gem "bullet_train-themes", BULLET_TRAIN_VERSION
+gem "bullet_train-themes-light", BULLET_TRAIN_VERSION
+gem "bullet_train-integrations", BULLET_TRAIN_VERSION
+gem "bullet_train-integrations-stripe", BULLET_TRAIN_VERSION
 
 # Optional support packages.
-gem "bullet_train-sortable"
-gem "bullet_train-scope_questions"
-gem "bullet_train-obfuscates_id"
+gem "bullet_train-sortable", BULLET_TRAIN_VERSION
+gem "bullet_train-scope_questions", BULLET_TRAIN_VERSION
+gem "bullet_train-obfuscates_id", BULLET_TRAIN_VERSION
+
+# Core gems that are dependencies of gems listed above. Technically they
+# shouldn't need to be listed here, but we list them so that we can keep
+# verion numbers in sync.
+gem "bullet_train-fields", BULLET_TRAIN_VERSION
```

Comment:
Can't the gems that depend on them specify the version in their gemspec?

Reply:
It seems safer that it's in the Gemfile rather than locked in a gemspec. In case something goes wrong and people need a fix where they can't downgrade, they'd need to fork and revise the gemspec version.

Of course, that's hypothetical and I don't know if we'd ever need this. Having the lever seems better to me though.

Reply:
I'm not positive if this would work, but I think if you explicitly set a version in the Gemfile it may override the version defined in the gemspec. That way you could downgrade if you really needed to.

Reply:
Just a thought...
changing this line

https://github.com/bullet-train-co/bullet_train-core/blob/13368d59430d45d25a907f79ad3b29bd865f6b60/bullet_train/bullet_train.gemspec#L36

to something like...

```
  spec.add_dependency "bullet_train-fields", BulletTrain::VERSION
```

Might help.  This would ensure that the bullet_train gem depends on the same version of the related gems. These are all now being published in lock-step now, so I don't think it would be a problem.

Reply:
I don't think that there's a way to do that in the various `.gemspec` files that wouldn't result in the problem that @kaspth described.

Like if for some (extremely weird) reason someone wanted to be on version `1.3.25` for _most_ of the BT gems, but wanted to be on `1.3.24` for `bullet_train-fields` they'd have no easy way to do that.

In order to keep things in sync `bullet_train` would have to declare a hard dependency on `bullet_train-fields` for _exactly_ version `1.3.25` and if you asked for `1.3.24` of `bullet_train-fields` in your `Gemfile` then `bundler` would complain at you about incompatible versions.

I think we want to leave the version requirements in `bullet_train.gemspec` as loose as possible and leave it up to the "consuming app" (aka the starter repo) to manage explicit version dependencies.